src/plugins/dashboard/public/services/types.ts

```diff
@@ -74,6 +74,6 @@ export interface DashboardServices {
   savedObjectsManagement: SavedObjectsManagementPluginStart;
   contentManagement: ContentManagementPublicStart;
   serverless: DashboardServerlessService; // TODO: make this optional in follow up
-  noDataPage: NoDataPageService;
+  noDataPage?: NoDataPageService;
```

Comment:
Based on other lines of code in this file, it looks like it's preferred to make plugin dependencies optional - so that's what I did as I was working in this area.

Reply:
Looks like this is throwing a TS error in `src/plugins/dashboard/public/plugin_services.ts` - because of the typing of `PluginServiceProviders`, we currently don't support optional services. We had an [issue](https://github.com/elastic/kibana/issues/140433) to address this, but we closed it in favour of a larger refactor - in the mean time, the best workaround for making a service "optional" is to make the inner properties optional instead.

Sorry about that - we know our services layer is a bit frustrating to work with ATM 🙈

Reply:
Thanks for the guidance @Heenawter! Fixed in f9395a9d6f3